server/src/main/java/de/uftos/services/TimetableService.java

```diff
@@ -276,28 +278,74 @@ private List<GradeProblemDto> getGrades() {
   private List<LessonProblemDto> getLessons(Timetable timetable) {
     //initializing new Lessons
     List<Curriculum> curriculums = curriculumRepository.findAll();
-    for (StudentGroup studentGroup : studentGroupRepository.findAll()) {
+    if (curriculums.isEmpty()) {
```

Comment:
I don't think the checks for curricula/lesson-counts/student-groups are necessary.